**OS path**
- 학원 노트북용 --> C:/Users/Playdata/Playdata_anl/1st_Project/dataset/population/total_final
- 개인 노트북용 --> C:/Users/Brian/Desktop/playdata/Playdata_anl/1st_Project/dataset/population/total_final

In [17]:
import pandas as pd
import numpy as np

- csv 파일을 불러올 때, **dtype 옵션**으로 mixed type으로 되어 있는 **date 변수를 str 타입으로 지정**해주자!!

In [18]:
pop_total_final = pd.read_csv("C:/Users/Playdata/Playdata_anl/1st_Project/dataset/population/total_final/pop_total_final.csv", dtype = {'date': str})
pop_total_final

Unnamed: 0      date  hour  dong_code       total      20세미만  \
0                  0  20170101     0   11110515  16846.9908  2459.9117   
1                  1  20170101     0   11110530  25606.3679  3318.1364   
2                  2  20170101     0   11110540   5724.9509   568.1162   
3                  3  20170101     0   11110550  11691.6393  1674.9696   
4                  4  20170101     0   11110560  24976.4805  3340.9352   
...              ...       ...   ...        ...         ...        ...   
11020603    11020603  20191231    23   11740610  47108.0384  6188.2727   
11020604    11020604  20191231    23   11740685  56004.5812  7727.9161   
11020605    11020605  20191231    23   11740640  26471.4234  4430.1550   
11020606    11020606  20191231    23   11740650  27986.5462  3526.7447   
11020607    11020607  20191231    23   11740700  32974.4339  5135.1512   

                20대        30대        40대        50대        60대      70세이상  
0         3071.7187  2619.6219  3120.0414  2147.3964  1387.4667  2040.8343  
1         4419.9298  4219.9963  4768.6949  3627.0648  2353.1643  2899.3791  
2         1348.8877   929.6296   953.8525   677.0148   523.6464   723.8041  
3         1339.5106  1371.8571  1914.1036  1769.2925  1489.8167  2132.0894  
4         2659.3631  2822.5062  3880.1131  4217.1843  3331.8599  4724.5195  
...             ...        ...        ...        ...        ...        ...  
11020603  6288.1413  7824.7680  7344.8905  7573.8382  6292.1836  5595.9437  
11020604  7064.1564  8598.2631  9226.5051  9408.6981  7522.9479  6456.0969  
11020605  2648.8209  4147.1218  4493.7940  4208.8375  3431.4259  3111.2694  
11020606  3996.3412  4789.6924  4511.4253  4309.2844  3528.7525  3324.3063  
11020607  3596.6190  4412.7418  4875.9918  5253.7418  4830.2941  4869.8948  

[11020608 rows x 12 columns]

In [19]:
pop_total_final.drop(['Unnamed: 0'], axis = 1, inplace = True) # 첫 번째 컬럼 제거(의미 없는 컬럼임)
pop_total_final.head()

date  hour  dong_code       total      20세미만        20대        30대  \
0  20170101     0   11110515  16846.9908  2459.9117  3071.7187  2619.6219   
1  20170101     0   11110530  25606.3679  3318.1364  4419.9298  4219.9963   
2  20170101     0   11110540   5724.9509   568.1162  1348.8877   929.6296   
3  20170101     0   11110550  11691.6393  1674.9696  1339.5106  1371.8571   
4  20170101     0   11110560  24976.4805  3340.9352  2659.3631  2822.5062   

         40대        50대        60대      70세이상  
0  3120.0414  2147.3964  1387.4667  2040.8343  
1  4768.6949  3627.0648  2353.1643  2899.3791  
2   953.8525   677.0148   523.6464   723.8041  
3  1914.1036  1769.2925  1489.8167  2132.0894  
4  3880.1131  4217.1843  3331.8599  4724.5195

### 노가다 작업 시작...

In [20]:
# len(set(pop_total_final['dong_code']))

In [21]:
# set(pop_total_final['dong_code'])

- **'11620685':'관악구 신사동'**는 딕셔너리에서 제외시켰다.
    - 왜냐하면 지하철 역 데이터 중, '신사동'은 '강남구'에만 존재하기 때문!!

In [22]:
dic = {'11110515':'종로구 청운효자동', '11110530':'종로구 사직동', '11110540':'종로구 삼청동', '11110550':'종로구 부암동', 
       '11110560':'종로구 평창동', '11110570':'종로구 무악동', '11110580':'종로구 교남동', '11110600':'종로구 가회동', 
       '11110615':'종로구 종로1.2.3.4가동', '11110630':'종로구 종로5.6가동', '11110640':'종로구 이화동', '11110650':'종로구 혜화동', 
       '11110670':'종로구 창신제1동', '11110680':'종로구 창신제2동', '11110690':'종로구 창신제3동', '11110700':'종로구 숭인제1동', 
       '11110710':'종로구 숭인제2동', 
       
       '11140520':'중구 소공동', '11140540':'중구 회현동', '11140550':'중구 명동', '11140570':'중구 필동', '11140580':'중구 장충동', 
       '11140590':'중구 광희동', '11140605':'중구 을지로동', '11140615':'중구 신당동', '11140625':'중구 다산동', '11140635':'중구 약수동', 
       '11140645':'중구 청구동', '11140650':'중구 신당제5동', '11140665':'중구 동화동', '11140670':'중구 황학동', '11140680':'중구 중림동', 
       
       '11170510':'용산구 후암동', '11170520':'용산구 용산2가동', '11170530':'용산구 남영동', '11170555':'용산구 청파동', 
       '11170560':'용산구 원효로제1동', '11170570':'용산구 원효로제2동', '11170580':'용산구 효창동', '11170590':'용산구 용문동', 
       '11170625':'용산구 한강로동', '11170630':'용산구 이촌제1동', '11170640':'용산구 이촌제2동', '11170650':'용산구 이태원제1동', 
       '11170660':'용산구 이태원제2동', '11170685':'용산구 한남동', '11170690':'용산구 서빙고동', '11170700':'용산구 보광동',
       
       '11200520':'성동구 왕십리제2동', '11200535':'성동구 왕십리도선동', '11200540':'성동구 마장동', '11200550':'성동구 사근동',
       '11200560':'성동구 행당제1동', '11200570':'성동구 행당제2동', '11200580':'성동구 응봉동', '11200590':'성동구 금호1가동',
       '11200615':'성동구 금호2.3가동', '11200620':'성동구 금호4가동', '11200645':'성동구 옥수동', '11200650':'성동구 성수1가제1동', 
       '11200660':'성동구 성수1가제2동', '11200670':'성동구 성수2가제1동', '11200690':'성동구 성수2가제3동', '11200720':'성동구 송정동', 
       '11200790':'성동구 용답동',
       
       '11215710':'광진구 화양동', '11215730':'광진구 군자동', '11215740':'광진구 중곡제1동', '11215750':'광진구 중곡제2동',
       '11215760':'광진구 중곡제3동', '11215770':'광진구 중곡제4동', '11215780':'광진구 능동', '11215810':'광진구 광장동',
       '11215820':'광진구 자양제1동', '11215830':'광진구 자양제2동', '11215840':'광진구 자양제3동', '11215847':'광진구 자양제4동',
       '11215850':'광진구 구의제1동', '11215860':'광진구 구의제2동', '11215870':'광진구 구의제3동',
       
       '11230536':'동대문구 용신동', '11230545':'동대문구 제기동', '11230560':'동대문구 전농제1동', '11230570':'동대문구 전농제2동',
       '11230600':'동대문구 답십리제1동', '11230610':'동대문구 답십리제2동', '11230650':'동대문구 장안제1동', '11230660':'동대문구 장안제2동',
       '11230705':'동대문구 청량리동', '11230710':'동대문구 회기동', '11230720':'동대문구 휘경제1동', '11230730':'동대문구 휘경제2동',
       '11230740':'동대문구 이문제1동', '11230750':'동대문구 이문제2동',
      
       '11260520':'중랑구 면목제2동', '11260540':'중랑구 면목제4동', '11260550':'중랑구 면목제5동', '11260565':'중랑구 면목본동',
       '11260570':'중랑구 면목제7동', '11260575':'중랑구 면목제3.8동', '11260580':'중랑구 상봉제1동', '11260590':'중랑구 상봉제2동', 
       '11260600':'중랑구 중화제1동', '11260610':'중랑구 중화제2동', '11260620':'중랑구 묵제1동', '11260630':'중랑구 묵제2동', 
       '11260655':'중랑구 망우본동', '11260660':'중랑구 망우제3동', '11260680':'중랑구 신내1동', '11260690':'중랑구 신내2동',
      
       '11290525':'성북구 성북동', '11290555':'성북구 삼선동', '11290575':'성북구 동선동', '11290580':'성북구 돈암제1동', 
       '11290590':'성북구 돈암제2동', '11290600':'성북구 안암동', '11290610':'성북구 보문동', '11290620':'성북구 정릉제1동', 
       '11290630':'성북구 정릉제2동', '11290640':'성북구 정릉제3동', '11290650':'성북구 정릉제4동', '11290660':'성북구 길음제1동', 
       '11290685':'성북구 길음제2동', '11290705':'성북구 종암동', '11290715':'성북구 월곡제1동', '11290725':'성북구 월곡제2동', 
       '11290760':'성북구 장위제1동', '11290770':'성북구 장위제2동', '11290780':'성북구 장위제3동', '11290810':'성북구 석관동',
       
       '11305534':'강북구 삼양동', '11305535':'강북구 미아동', '11305545':'강북구 송중동', '11305555':'강북구 송천동',
       '11305575':'강북구 삼각산동', '11305590':'강북구 번제1동', '11305600':'강북구 번제2동', '11305606':'강북구 번제3동',
       '11305610':'강북구 수유제1동', '11305620':'강북구 수유제2동', '11305630':'강북구 수유제3동', '11305645':'강북구 우이동', 
       '11305660':'강북구 인수동',
      
       '11320511':'도봉구 창제1동', '11320512':'도봉구 창제2동', '11320513':'도봉구 창제3동', '11320514':'도봉구 창제4동',
       '11320515':'도봉구 창제5동', '11320521':'도봉구 도봉제1동', '11320522':'도봉구 도봉제2동', '11320660':'도봉구 쌍문제1동',
       '11320670':'도봉구 쌍문제2동', '11320680':'도봉구 쌍문제3동', '11320681':'도봉구 쌍문제4동', '11320690':'도봉구 방학제1동',
       '11320700':'도봉구 방학제2동', '11320710':'도봉구 방학제3동',
      
       '11350560':'노원구 월계1동', '11350570':'노원구 월계2동', '11350580':'노원구 월계3동', '11350595':'노원구 공릉1동',
       '11350600':'노원구 공릉2동', '11350611':'노원구 하계1동', '11350612':'노원구 하계2동', '11350619':'노원구 중계본동',
       '11350621':'노원구 중계1동', '11350624':'노원구 중계4동', '11350625':'노원구 중계2.3동', '11350630':'노원구 상계1동', 
       '11350640':'노원구 상계2동', '11350665':'노원구 상계3.4동', '11350670':'노원구 상계5동', '11350695':'노원구 상계6.7동', 
       '11350700':'노원구 상계8동', '11350710':'노원구 상계9동', '11350720':'노원구 상계10동',
      
       '11380510':'은평구 녹번동', '11380520':'은평구 불광제1동', '11380530':'은평구 불광제2동', '11380551':'은평구 갈현제1동',
       '11380552':'은평구 갈현제2동', '11380560':'은평구 구산동', '11380570':'은평구 대조동', '11380580':'은평구 응암제1동',
       '11380590':'은평구 응암제2동', '11380600':'은평구 응암제3동', '11380625':'은평구 역촌동', '11380631':'은평구 신사제1동',
       '11380632':'은평구 신사제2동', '11380640':'은평구 증산동', '11380650':'은평구 수색동', '11380690':'은평구 진관동',
      
       '11410520':'서대문구 천연동', '11410555':'서대문구 북아현동', '11410565':'서대문구 충현동', '11410585':'서대문구 신촌동',
       '11410615':'서대문구 연희동', '11410620':'서대문구 홍제제1동', '11410640':'서대문구 홍제제3동', '11410655':'서대문구 홍제제2동',
       '11410660':'서대문구 홍은제1동', '11410685':'서대문구 홍은제2동', '11410690':'서대문구 남가좌제1동', '11410700':'서대문구 남가좌제2동',
       '11410710':'서대문구 북가좌제1동', '11410720':'서대문구 북가좌제2동',
      
       '11440555':'마포구 아현동', '11440565':'마포구 공덕동', '11440585':'마포구 도화동', '11440590':'마포구 용강동', 
       '11440600':'마포구 대흥동', '11440610':'마포구 염리동', '11440630':'마포구 신수동', '11440655':'마포구 서강동',
       '11440660':'마포구 서교동', '11440680':'마포구 합정동', '11440690':'마포구 망원제1동', '11440700':'마포구 망원제2동',
       '11440710':'마포구 연남동', '11440720':'마포구 성산제1동', '11440730':'마포구 성산제2동', '11440740':'마포구 상암동',
      
       '11470510':'양천구 목1동', '11470520':'양천구 목2동', '11470530':'양천구 목3동', '11470540':'양천구 목4동',
       '11470550':'양천구 목5동', '11470560':'양천구 신월1동', '11470570':'양천구 신월2동', '11470580':'양천구 신월3동',
       '11470590':'양천구 신월4동', '11470600':'양천구 신월5동', '11470610':'양천구 신월6동', '11470611':'양천구 신월7동',
       '11470620':'양천구 신정1동', '11470630':'양천구 신정2동', '11470640':'양천구 신정3동', '11470650':'양천구 신정4동',
       '11470670':'양천구 신정6동', '11470680':'양천구 신정7동',
      
       '11500510':'강서구 염창동', '11500520':'강서구 등촌제1동', '11500530':'강서구 등촌제2동', '11500535':'강서구 등촌제3동',
       '11500540':'강서구 화곡제1동', '11500550':'강서구 화곡제2동', '11500560':'강서구 화곡제3동', '11500570':'강서구 화곡제4동',
       '11500590':'강서구 화곡본동', '11500591':'강서구 화곡제6동', '11500593':'강서구 화곡제8동', '11500603':'강서구 가양제1동',
       '11500604':'강서구 가양제2동', '11500605':'강서구 가양제3동', '11500611':'강서구 발산제1동', '11500615':'강서구 우장산동',
       '11500620':'강서구 공항동', '11500630':'강서구 방화제1동', '11500640':'강서구 방화제2동', '11500641':'강서구 방화제3동',
      
       '11530510':'구로구 신도림동', '11530520':'구로구 구로제1동', '11530530':'구로구 구로제2동', '11530540':'구로구 구로제3동',
       '11530550':'구로구 구로제4동', '11530560':'구로구 구로제5동', '11530595':'구로구 가리봉동', '11530720':'구로구 고척제1동',
       '11530730':'구로구 고척제2동', '11530740':'구로구 개봉제1동', '11530750':'구로구 개봉제2동', '11530760':'구로구 개봉제3동',
       '11530770':'구로구 오류제1동', '11530780':'구로구 오류제2동', '11530790':'구로구 수궁동',
      
       '11545510':'금천구 가산동', '11545610':'금천구 독산제1동', '11545620':'금천구 독산제2동', '11545630':'금천구 독산제3동',
       '11545640':'금천구 독산제4동', '11545670':'금천구 시흥제1동', '11545680':'금천구 시흥제2동', '11545690':'금천구 시흥제3동',
       '11545700':'금천구 시흥제4동', '11545710':'금천구 시흥제5동',
      
       '11560515':'영등포구 영등포본동', '11560535':'영등포구 영등포동', '11560540':'영등포구 여의동', '11560550':'영등포구 당산제1동',
       '11560560':'영등포구 당산제2동', '11560585':'영등포구 도림동', '11560605':'영등포구 문래동', '11560610':'영등포구 양평제1동',
       '11560620':'영등포구 양평제2동', '11560630':'영등포구 신길제1동', '11560650':'영등포구 신길제3동', '11560660':'영등포구 신길제4동',
       '11560670':'영등포구 신길제5동', '11560680':'영등포구 신길제6동', '11560690':'영등포구 신길제7동', '11560700':'영등포구 대림제1동',
       '11560710':'영등포구 대림제2동', '11560720':'영등포구 대림제3동',
      
       '11590510':'동작구 노량진제1동', '11590520':'동작구 노량진제2동', '11590530':'동작구 상도제1동', '11590540':'동작구 상도제2동',
       '11590550':'동작구 상도제3동', '11590560':'동작구 상도제4동', '11590605':'동작구 흑석동', '11590620':'동작구 사당제1동',
       '11590630':'동작구 사당제2동', '11590640':'동작구 사당제3동', '11590650':'동작구 사당제4동', '11590651':'동작구 사당제5동',
       '11590660':'동작구 대방동', '11590670':'동작구 신대방제1동', '11590680':'동작구 신대방제2동',
      
       '11620525':'관악구 보라매동', '11620545':'관악구 청림동', '11620565':'관악구 성현동', '11620575':'관악구 행운동',
       '11620585':'관악구 낙성대동', '11620595':'관악구 청룡동', '11620605':'관악구 은천동', '11620615':'관악구 중앙동',
       '11620625':'관악구 인헌동', '11620630':'관악구 남현동', '11620645':'관악구 서원동', '11620655':'관악구 신원동',
       '11620665':'관악구 서림동', '11620695':'관악구 신림동', '11620715':'관악구 난향동', '11620725':'관악구 조원동', 
       '11620735':'관악구 대학동', '11620745':'관악구 삼성동', '11620765':'관악구 미성동', '11620775':'관악구 난곡동',
      
       '11650510':'서초구 서초1동', '11650520':'서초구 서초2동', '11650530':'서초구 서초3동', '11650531':'서초구 서초4동',
       '11650540':'서초구 잠원동', '11650550':'서초구 반포본동', '11650560':'서초구 반포1동', '11650570':'서초구 반포2동',
       '11650580':'서초구 반포3동', '11650581':'서초구 반포4동', '11650590':'서초구 방배본동', '11650600':'서초구 방배1동',
       '11650610':'서초구 방배2동', '11650620':'서초구 방배3동', '11650621':'서초구 방배4동', '11650651':'서초구 양재1동',
       '11650652':'서초구 양재2동', '11650660':'서초구 내곡동',
      
       '11680510':'강남구 신사동', '11680521':'강남구 논현1동', '11680531':'강남구 논현2동', '11680545':'강남구 압구정동',
       '11680565':'강남구 청담동', '11680580':'강남구 삼성1동', '11680590':'강남구 삼성2동', '11680600':'강남구 대치1동',
       '11680610':'강남구 대치2동', '11680630':'강남구 대치4동', '11680640':'강남구 역삼1동', '11680650':'강남구 역삼2동',
       '11680655':'강남구 도곡1동', '11680656':'강남구 도곡2동', '11680660':'강남구 개포1동', '11680670':'강남구 개포2동',
       '11680690':'강남구 개포4동', '11680700':'강남구 세곡동', '11680720':'강남구 일원본동', '11680730':'강남구 일원1동',
       '11680740':'강남구 일원2동', '11680750':'강남구 수서동',
      
       '11710510':'송파구 풍납1동', '11710520':'송파구 풍납2동', '11710531':'송파구 거여1동', '11710532':'송파구 거여2동',
       '11710540':'송파구 마천1동', '11710550':'송파구 마천2동', '11710561':'송파구 방이1동', '11710562':'송파구 방이2동',
       '11710566':'송파구 오륜동', '11710570':'송파구 오금동', '11710580':'송파구 송파1동', '11710590':'송파구 송파2동',
       '11710600':'송파구 석촌동', '11710610':'송파구 삼전동', '11710620':'송파구 가락본동', '11710631':'송파구 가락1동',
       '11710632':'송파구 가락2동', '11710641':'송파구 문정1동', '11710642':'송파구 문정2동', '11710646':'송파구 장지동',
       '11710647':'송파구 위례동', '11710650':'송파구 잠실본동', '11710670':'송파구 잠실2동', '11710680':'송파구 잠실3동', 
       '11710690':'송파구 잠실4동', '11710710':'송파구 잠실6동', '11710720':'송파구 잠실7동',
      
       '11740515':'강동구 강일동', '11740520':'강동구 상일동', '11740530':'강동구 명일제1동', '11740540':'강동구 명일제2동',
       '11740550':'강동구 고덕제1동', '11740560':'강동구 고덕제2동', '11740570':'강동구 암사제1동', '11740580':'강동구 암사제2동',
       '11740590':'강동구 암사제3동', '11740600':'강동구 천호제1동', '11740610':'강동구 천호제2동', '11740620':'강동구 천호제3동',
       '11740640':'강동구 성내제1동', '11740650':'강동구 성내제2동', '11740660':'강동구 성내제3동', '11740685':'강동구 길동',
       '11740690':'강동구 둔촌제1동', '11740700':'강동구 둔촌제2동'}

**dictionary를 DataFrame으로 변경**

In [23]:
dic_df = pd.DataFrame.from_dict(dic, orient = 'index')
dic_df = dic_df.reset_index()
dic_df.rename(columns = {'index':'dong_code', 0:'dong'}, inplace = True)
dic_df

dong_code       dong
0    11110515  종로구 청운효자동
1    11110530    종로구 사직동
2    11110540    종로구 삼청동
3    11110550    종로구 부암동
4    11110560    종로구 평창동
..        ...        ...
418  11740650  강동구 성내제2동
419  11740660  강동구 성내제3동
420  11740685     강동구 길동
421  11740690  강동구 둔촌제1동
422  11740700  강동구 둔촌제2동

[423 rows x 2 columns]

- dong 변수를 "종로구_청운효자동" 이런 식으로 변경해주겠다.

In [24]:
result = []
for i in range(len(dic_df)):
    dong_split = dic_df['dong'][i].split()
    result.append('_'.join(dong_split))
    
result

['종로구_청운효자동',
 '종로구_사직동',
 '종로구_삼청동',
 '종로구_부암동',
 '종로구_평창동',
 '종로구_무악동',
 '종로구_교남동',
 '종로구_가회동',
 '종로구_종로1.2.3.4가동',
 '종로구_종로5.6가동',
 '종로구_이화동',
 '종로구_혜화동',
 '종로구_창신제1동',
 '종로구_창신제2동',
 '종로구_창신제3동',
 '종로구_숭인제1동',
 '종로구_숭인제2동',
 '중구_소공동',
 '중구_회현동',
 '중구_명동',
 '중구_필동',
 '중구_장충동',
 '중구_광희동',
 '중구_을지로동',
 '중구_신당동',
 '중구_다산동',
 '중구_약수동',
 '중구_청구동',
 '중구_신당제5동',
 '중구_동화동',
 '중구_황학동',
 '중구_중림동',
 '용산구_후암동',
 '용산구_용산2가동',
 '용산구_남영동',
 '용산구_청파동',
 '용산구_원효로제1동',
 '용산구_원효로제2동',
 '용산구_효창동',
 '용산구_용문동',
 '용산구_한강로동',
 '용산구_이촌제1동',
 '용산구_이촌제2동',
 '용산구_이태원제1동',
 '용산구_이태원제2동',
 '용산구_한남동',
 '용산구_서빙고동',
 '용산구_보광동',
 '성동구_왕십리제2동',
 '성동구_왕십리도선동',
 '성동구_마장동',
 '성동구_사근동',
 '성동구_행당제1동',
 '성동구_행당제2동',
 '성동구_응봉동',
 '성동구_금호1가동',
 '성동구_금호2.3가동',
 '성동구_금호4가동',
 '성동구_옥수동',
 '성동구_성수1가제1동',
 '성동구_성수1가제2동',
 '성동구_성수2가제1동',
 '성동구_성수2가제3동',
 '성동구_송정동',
 '성동구_용답동',
 '광진구_화양동',
 '광진구_군자동',
 '광진구_중곡제1동',
 '광진구_중곡제2동',
 '광진구_중곡제3동',
 '광진구_중곡제4동',
 '광진구_능동',
 '광진구_광장동',
 '광진구_자양제1동',
 '광진구_자양제2동',
 '광진구_자양제3동',
 '광진구_자양제4동

In [25]:
dic_df['dong'] = result
dic_df

dong_code       dong
0    11110515  종로구_청운효자동
1    11110530    종로구_사직동
2    11110540    종로구_삼청동
3    11110550    종로구_부암동
4    11110560    종로구_평창동
..        ...        ...
418  11740650  강동구_성내제2동
419  11740660  강동구_성내제3동
420  11740685     강동구_길동
421  11740690  강동구_둔촌제1동
422  11740700  강동구_둔촌제2동

[423 rows x 2 columns]

- 두 데이터 셋의 dong_code 변수 타입이 서로 다르다.
- 때문에 dic_df 데이터 셋의 dong_code 변수를 int 타입으로 변경해주겠다.

In [26]:
pop_total_final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11020608 entries, 0 to 11020607
Data columns (total 11 columns):
 #   Column     Dtype  
---  ------     -----  
 0   date       object 
 1   hour       int64  
 2   dong_code  int64  
 3   total      float64
 4   20세미만      float64
 5   20대        float64
 6   30대        float64
 7   40대        float64
 8   50대        float64
 9   60대        float64
 10  70세이상      float64
dtypes: float64(8), int64(2), object(1)
memory usage: 924.9+ MB


In [27]:
dic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 423 entries, 0 to 422
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   dong_code  423 non-null    object
 1   dong       423 non-null    object
dtypes: object(2)
memory usage: 6.7+ KB


In [28]:
dic_df['dong_code'] = pd.to_numeric(dic_df['dong_code'])
dic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 423 entries, 0 to 422
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   dong_code  423 non-null    int64 
 1   dong       423 non-null    object
dtypes: int64(1), object(1)
memory usage: 6.7+ KB


- pop_total_final 데이터 셋에는 dong_code가 424개가 존재하는데, dic_df 데이터 셋에는 dong_code가 423개 존재한다.
- 최종 inner join한 데이터 셋을 보면, 1개 차이로 인해 len이 불일치한다..
- 따라서 이 부분을 해결해주고 inner join을 수행해주도록 하겠다.
    - 해결 방안은 직접 비교해서 dic_df 데이터 셋에 미존재하는 **"11710647(00)"** 행정동을 직접 찾아서 넣어주어야 한다.
    - 구글링을 해 본 결과, **"11710647(00)"**에 해당하는 행정동은 **"위례동"**임을 확인했다.
    - 그래서 dic(노가다 작업해 준 딕셔너리)에 **```'11710647':'위례동'```** 코드를 넣어주었다!!

In [29]:
# df1 = pd.DataFrame(set(pop_total_final['dong_code']))
# df2 = pd.DataFrame(set(dic_df['dong_code']))

# df1.rename(columns = {0:'dong_code'}, inplace = True)
# df2.rename(columns = {0:'dong_code'}, inplace = True)

# check1 = df1.sort_values(by = 'dong_code')
# check2 = df2.sort_values(by = 'dong_code')

In [30]:
# # c_total 데이터 셋을 50개의 row 씩 잘라서 눈으로 확인해보았음
# # 그 결과, dic_df 데이터 셋에 "11710647" 행정동코드가 미존재함을 확인하였음!!
# c_total = pd.concat([check1, check2], axis = 1)

In [31]:
# pop_total_final[pop_total_final['dong_code'] == 11710647]

**pop_total_final 데이터 셋과 dic_df 데이터 셋을 inner join**

In [32]:
pop_dong_total_final = pd.merge(pop_total_final, dic_df, how = 'inner', on = 'dong_code')
pop_dong_total_final

date  hour  dong_code       total      20세미만        20대  \
0         20170101     0   11110515  16846.9908  2459.9117  3071.7187   
1         20170101     1   11110515  14625.3361  2319.8398  2065.8258   
2         20170101     2   11110515  14463.5458  2317.3200  2009.3034   
3         20170101     3   11110515  14590.0305  2346.4891  2031.8727   
4         20170101     4   11110515  14768.4378  2330.3941  2068.9656   
...            ...   ...        ...         ...        ...        ...   
10994611  20191231    19   11740700  28966.0305  4402.7464  2984.5773   
10994612  20191231    20   11740700  29495.9850  4460.1392  3123.6626   
10994613  20191231    21   11740700  30131.0225  4616.2185  3210.7351   
10994614  20191231    22   11740700  31575.9329  5043.9562  3474.2488   
10994615  20191231    23   11740700  32974.4339  5135.1512  3596.6190   

                30대        40대        50대        60대      70세이상       dong  
0         2619.6219  3120.0414  2147.3964  1387.4667  2040.8343  종로구_청운효자동  
1         2172.7630  2725.2784  2005.3089  1324.3817  2011.9376  종로구_청운효자동  
2         2145.7461  2703.1343  1955.3093  1312.1868  2020.5454  종로구_청운효자동  
3         2189.7287  2698.9883  1974.9110  1323.8786  2024.1609  종로구_청운효자동  
4         2222.9972  2699.3068  1991.6161  1368.5205  2086.6376  종로구_청운효자동  
...             ...        ...        ...        ...        ...        ...  
10994611  3907.6240  4294.0431  4656.8790  4323.5792  4396.5802  강동구_둔촌제2동  
10994612  3978.6890  4387.9537  4781.8335  4324.7054  4439.0004  강동구_둔촌제2동  
10994613  4093.1960  4452.0349  4850.2092  4426.8812  4481.7475  강동구_둔촌제2동  
10994614  4300.4638  4637.4441  5022.1798  4576.0351  4521.6059  강동구_둔촌제2동  
10994615  4412.7418  4875.9918  5253.7418  4830.2941  4869.8948  강동구_둔촌제2동  

[10994616 rows x 12 columns]

In [33]:
len(pop_total_final)

11020608

- 위 결과를 보면, inner join이 잘 이루어진 것을 확인할 수 있다.

- 추후에 지하철 데이터와 결합할 때 조인키로 사용할 **pop_join_key 변수**를 생성해주겠다.
- 그 전에, **date 변수**에 **'20181130"'이라는 값을 '20181130'으로 변경**해주었다!!
    - 이 부분 때문에 pd.to_datetime()이 안 먹혔던 것이었다...

In [34]:
pop_dong_total_final[pop_dong_total_final['date'].str.len() > 8]

date  hour  dong_code       total     20세미만        20대  \
7112568  20181130"     0   11530780  47625.4881  9588.922  5422.5807   

               30대        40대        50대      60대      70세이상       dong  
7112568  7394.0737  8056.0807  6289.0262  5251.87  5622.9346  구로구_오류제2동

In [35]:
pop_dong_total_final['date'][7112568] = '20181130'

C:\Users\Playdata\anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [36]:
pop_dong_total_final['date'] = pd.to_datetime(pop_dong_total_final['date'], format = '%Y-%m-%d')
pop_dong_total_final['date']

0          2017-01-01
1          2017-01-01
2          2017-01-01
3          2017-01-01
4          2017-01-01
              ...    
10994611   2019-12-31
10994612   2019-12-31
10994613   2019-12-31
10994614   2019-12-31
10994615   2019-12-31
Name: date, Length: 10994616, dtype: datetime64[ns]

In [37]:
pop_dong_total_final.to_csv("C:/Users/Playdata/Playdata_anl/1st_Project/dataset/population/total_final/pop_dong_total_semi_final.csv")

In [38]:
pop_dong_total_final = pd.read_csv("C:/Users/Playdata/Playdata_anl/1st_Project/dataset/population/total_final/pop_dong_total_semi_final.csv", dtype = {'date': str})
pop_dong_total_final.drop(['Unnamed: 0'], axis = 1, inplace = True) # 첫 번째 컬럼 제거(의미 없는 컬럼임)

In [39]:
pop_dong_total_final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10994616 entries, 0 to 10994615
Data columns (total 12 columns):
 #   Column     Dtype  
---  ------     -----  
 0   date       object 
 1   hour       int64  
 2   dong_code  int64  
 3   total      float64
 4   20세미만      float64
 5   20대        float64
 6   30대        float64
 7   40대        float64
 8   50대        float64
 9   60대        float64
 10  70세이상      float64
 11  dong       object 
dtypes: float64(8), int64(2), object(2)
memory usage: 1006.6+ MB


In [40]:
pop_dong_total_final['pop_join_key'] = pop_dong_total_final['date'] + '_' + pop_dong_total_final['dong']
pop_dong_total_final.head()

date  hour  dong_code       total      20세미만        20대        30대  \
0  2017-01-01     0   11110515  16846.9908  2459.9117  3071.7187  2619.6219   
1  2017-01-01     1   11110515  14625.3361  2319.8398  2065.8258  2172.7630   
2  2017-01-01     2   11110515  14463.5458  2317.3200  2009.3034  2145.7461   
3  2017-01-01     3   11110515  14590.0305  2346.4891  2031.8727  2189.7287   
4  2017-01-01     4   11110515  14768.4378  2330.3941  2068.9656  2222.9972   

         40대        50대        60대      70세이상       dong          pop_join_key  
0  3120.0414  2147.3964  1387.4667  2040.8343  종로구_청운효자동  2017-01-01_종로구_청운효자동  
1  2725.2784  2005.3089  1324.3817  2011.9376  종로구_청운효자동  2017-01-01_종로구_청운효자동  
2  2703.1343  1955.3093  1312.1868  2020.5454  종로구_청운효자동  2017-01-01_종로구_청운효자동  
3  2698.9883  1974.9110  1323.8786  2024.1609  종로구_청운효자동  2017-01-01_종로구_청운효자동  
4  2699.3068  1991.6161  1368.5205  2086.6376  종로구_청운효자동  2017-01-01_종로구_청운효자동

### 최종 데이터 셋 csv 파일로 내보내기

In [42]:
pop_dong_total_final.to_csv("C:/Users/Playdata/Playdata_anl/1st_Project/dataset/population/total_final/pop_dong_total_final.csv")